/dashboard/settings 主题设置跟随系统存在bug

如图
<img width="1467" height="860" alt="Image" src="https://github.com/user-attachments/assets/50037653-16fe-4270-9abc-c3907671ca40" />

这里的锅，用户设置没有和系统设置同步

<img width="449" height="162" alt="Image" src="https://github.com/user-attachments/assets/66cf9dd4-4f5f-4ebc-aac8-35ed9c9696d3" />

但是设置成跟随系统提示保存设置失败
/api/users/me/preferences  500错误
```
{
    "error": {
        "type": "system_error",
        "message": "系统遇到未知错误，请稍后重试或联系管理员",
        "details": {
            "error_id": "b3a8912d",
            "recovery_info": "请稍后重试，如问题持续请联系管理员"
        }
    }
}
```
```
2025-12-17 17:34:45 | INFO     | AUDIT [request_failed] - PUT /api/users/me/preferences via base | user_id=76c60653-39ee-41b5-a098-8001b65f8c05, ip=192.168.1.253

2025-12-17 17:34:45 | ERROR    | 未知错误 [b3a8912d]: Invalid theme. Must be 'light', 'dark', or 'auto'

2025-12-17 17:34:45 | ERROR    | Error response: system_error - 系统遇到未知错误，请稍后重试或联系管理员

[2025-12-17 17:34:45 +0800] [44] [ERROR] Exception in ASGI application

Traceback (most recent call last):

  File "/usr/local/lib/python3.12/site-packages/uvicorn/protocols/http/httptools_impl.py", line 409, in run_asgi

    result = await app(  # type: ignore[func-returns-value]

             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.12/site-packages/uvicorn/middleware/proxy_headers.py", line 60, in __call__

    return await self.app(scope, receive, send)

           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.12/site-packages/fastapi/applications.py", line 1139, in __call__

    await super().__call__(scope, receive, send)

  File "/usr/local/lib/python3.12/site-packages/starlette/applications.py", line 107, in __call__

    await self.middleware_stack(scope, receive, send)

  File "/usr/local/lib/python3.12/site-packages/starlette/middleware/errors.py", line 186, in __call__

    raise exc

  File "/usr/local/lib/python3.12/site-packages/starlette/middleware/errors.py", line 164, in __call__

    await self.app(scope, receive, _send)

  File "/usr/local/lib/python3.12/site-packages/starlette/middleware/base.py", line 191, in __call__

    with recv_stream, send_stream, collapse_excgroups():

                                   ^^^^^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.12/contextlib.py", line 158, in __exit__

    self.gen.throw(value)

  File "/usr/local/lib/python3.12/site-packages/starlette/_utils.py", line 85, in collapse_excgroups

    raise exc

  File "/usr/local/lib/python3.12/site-packages/starlette/middleware/base.py", line 193, in __call__

    response = await self.dispatch_func(request, call_next)

               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/app/src/middleware/plugin_middleware.py", line 221, in dispatch

    raise exception_to_raise

  File "/app/src/middleware/plugin_middleware.py", line 109, in dispatch

    response = await call_next(request)

               ^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.12/site-packages/starlette/middleware/base.py", line 168, in call_next

    raise app_exc from app_exc.__cause__ or app_exc.__context__

  File "/usr/local/lib/python3.12/site-packages/starlette/middleware/base.py", line 144, in coro

    await self.app(scope, receive_or_disconnect, send_no_error)

  File "/usr/local/lib/python3.12/site-packages/starlette/middleware/exceptions.py", line 63, in __call__

    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)

  File "/usr/local/lib/python3.12/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app

    raise exc

  File "/usr/local/lib/python3.12/site-packages/starlette/_exception_handler.py", line 42, in wrapped_app

    await app(scope, receive, sender)

  File "/usr/local/lib/python3.12/site-packages/fastapi/middleware/asyncexitstack.py", line 18, in __call__

    await self.app(scope, receive, send)

  File "/usr/local/lib/python3.12/site-packages/starlette/routing.py", line 716, in __call__

    await self.middleware_stack(scope, receive, send)

  File "/usr/local/lib/python3.12/site-packages/starlette/routing.py", line 736, in app

    await route.handle(scope, receive, send)

  File "/usr/local/lib/python3.12/site-packages/starlette/routing.py", line 290, in handle

    await self.app(scope, receive, send)

  File "/usr/local/lib/python3.12/site-packages/fastapi/routing.py", line 120, in app

    await wrap_app_handling_exceptions(app, request)(scope, receive, send)

  File "/usr/local/lib/python3.12/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app

    raise exc

  File "/usr/local/lib/python3.12/site-packages/starlette/_exception_handler.py", line 42, in wrapped_app

    await app(scope, receive, sender)

  File "/usr/local/lib/python3.12/site-packages/fastapi/routing.py", line 106, in app

    response = await f(request)

               ^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.12/site-packages/fastapi/routing.py", line 430, in app

    raw_response = await run_endpoint_function(

                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/usr/local/lib/python3.12/site-packages/fastapi/routing.py", line 316, in run_endpoint_function

    return await dependant.call(**values)

           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/app/src/api/user_me/routes.py", line 188, in update_my_preferences

    return await pipeline.run(adapter=adapter, http_request=request, db=db, mode=adapter.mode)

           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/app/src/api/base/pipeline.py", line 103, in run

    response = await adapter.handle(context)

               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

  File "/app/src/api/user_me/routes.py", line 926, in handle

    PreferenceService.update_preferences(

  File "/app/src/services/user/preference.py", line 75, in update_preferences

    raise ValueError("Invalid theme. Must be 'light', 'dark', or 'auto'")

ValueError: Invalid theme. Must be 'light', 'dark', or 'auto'
```

--- a/frontend/src/views/user/Settings.vue
+++ b/frontend/src/views/user/Settings.vue
@@ -62,6 +62,7 @@
             <Button
               type="submit"
               :disabled="savingProfile"
+              class="shadow-none hover:shadow-none"
             >
               {{ savingProfile ? '保存中...' : '保存修改' }}
             </Button>
@@ -107,6 +108,7 @@
             <Button
               type="submit"
               :disabled="changingPassword"
+              class="shadow-none hover:shadow-none"
             >
               {{ changingPassword ? '修改中...' : '修改密码' }}
             </Button>
@@ -320,6 +322,7 @@
 import { ref, onMounted } from 'vue'
 import { useAuthStore } from '@/stores/auth'
 import { meApi, type Profile } from '@/api/me'
+import { useDarkMode, type ThemeMode } from '@/composables/useDarkMode'
 import Card from '@/components/ui/card.vue'
 import Button from '@/components/ui/button.vue'
 import Badge from '@/components/ui/badge.vue'
@@ -338,6 +341,7 @@ import { log } from '@/utils/logger'
 
 const authStore = useAuthStore()
 const { success, error: showError } = useToast()
+const { setThemeMode } = useDarkMode()
 
 const profile = ref<Profile | null>(null)
 
@@ -375,20 +379,8 @@ function handleThemeChange(value: string) {
   themeSelectOpen.value = false
   updatePreferences()
 
-  // 应用主题
-  if (value === 'dark') {
-    document.documentElement.classList.add('dark')
-  } else if (value === 'light') {
-    document.documentElement.classList.remove('dark')
-  } else {
-    // system: 跟随系统
-    const prefersDark = window.matchMedia('(prefers-color-scheme: dark)').matches
-    if (prefersDark) {
-      document.documentElement.classList.add('dark')
-    } else {
-      document.documentElement.classList.remove('dark')
-    }
-  }
+  // 使用 useDarkMode 统一切换主题
+  setThemeMode(value as ThemeMode)
 }
 
 function handleLanguageChange(value: string) {
@@ -418,10 +410,16 @@ async function loadProfile() {
 async function loadPreferences() {
   try {
     const prefs = await meApi.getPreferences()
+
+    // 主题以本地 localStorage 为准（useDarkMode 在应用启动时已初始化）
+    // 这样可以避免刷新页面时主题被服务端旧值覆盖
+    const { themeMode: currentThemeMode } = useDarkMode()
+    const localTheme = currentThemeMode.value
+
     preferencesForm.value = {
       avatar_url: prefs.avatar_url || '',
       bio: prefs.bio || '',
-      theme: prefs.theme || 'light',
+      theme: localTheme,  // 使用本地主题，而非服务端返回值
       language: prefs.language || 'zh-CN',
       timezone: prefs.timezone || 'Asia/Shanghai',
       notifications: {
@@ -431,11 +429,12 @@ async function loadPreferences() {
       }
     }
 
-    // 应用主题
-    if (preferencesForm.value.theme === 'dark') {
-      document.documentElement.classList.add('dark')
-    } else if (preferencesForm.value.theme === 'light') {
-      document.documentElement.classList.remove('dark')
+    // 如果本地主题和服务端不一致，同步到服务端（静默更新，不提示用户）
+    const serverTheme = prefs.theme || 'light'
+    if (localTheme !== serverTheme) {
+      meApi.updatePreferences({ theme: localTheme }).catch(() => {
+        // 静默失败，不影响用户体验
+      })
     }
   } catch (error) {
     log.error('加载偏好设置失败:', error)

--- a/src/services/user/preference.py
+++ b/src/services/user/preference.py
@@ -71,8 +71,8 @@ class PreferenceService:
                 raise NotFoundException("Provider not found or inactive")
             preferences.default_provider_id = default_provider_id
         if theme is not None:
-            if theme not in ["light", "dark", "auto"]:
-                raise ValueError("Invalid theme. Must be 'light', 'dark', or 'auto'")
+            if theme not in ["light", "dark", "auto", "system"]:
+                raise ValueError("Invalid theme. Must be 'light', 'dark', 'auto', or 'system'")
             preferences.theme = theme
         if language is not None:
             preferences.language = language

